spec/models/otp_request_spec.rb

```diff
@@ -65,7 +65,7 @@
     context 'When otp is of length greater than 6 digits' do
       it 'should not be valid' do
         invalid_otp = '300099282'
```

Comment:
Let's move repeated values to `let` blocks. This will make our tests cleaner and easier to change. We can set up `valid_otp`, `invalid_mobile_number` and `expire_at` as shared values.

Also using `let` instead of creating values in each test can help with test speed (but the effect is small here). 
`let` is lazy-loaded, so it only runs when needed. This can save time if the value isn't used in every test.
But for simple values like these, the time saved is very little. The main benefit is making the code cleaner and easier to change.

Reply:
Specs are updated to use shoulda_matchers. Resolving this comment. Please feel free to re-open it if we have anything more to discuss here